internal/controller/persistentvolumeclaim_controller.go

```diff
@@ -66,19 +57,88 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		}, nil
 	}
 
-	// Skip if the PVC is not deleted or PVC does not have TopoLVM's finalizer.
-	if pvc.DeletionTimestamp == nil {
+	// Skip if PVC does not have TopoLVM's finalizer which added by the Pod mutating webhook.
+	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
 		return ctrl.Result{}, nil
 	}
 
-	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
-		return ctrl.Result{}, nil
+	if pvc.DeletionTimestamp == nil {
+		err := r.notifyKubeletToResizeFS(ctx, pvc)
+		return ctrl.Result{}, err
+	}
+
+	result, err := r.deletePodsUsingDeletingPVC(ctx, pvc)
+	if result.Requeue || err != nil {
+		return result, err
+	}
+
+	updated := controllerutil.RemoveFinalizer(pvc, topolvm.PVCFinalizer)
+	if updated {
+		if err := r.client.Update(ctx, pvc); err != nil {
+			log.Error(err, "failed to remove finalizer", "name", pvc.Name)
```

Comment:
```suggestion
			log.Error(err, "failed to remove finalizer")
```

For the same reason as 78868e29723289704fdf173fe0380865819d4176.